### PR TITLE
Updating Select theme doc.js to correctly reference `select.options.box` instead of `select.options.container`

### PR DESCRIPTION
**Pull request by halocline**
Signed-off-by: Matt Glissmann <mdglissmann@gmail.com>

#### What does this PR do?
Updates `Select/doc.js` to reference `select.options.box' instead of `select.options.container` which does not exist in `themes/base.js`.

#### Where should the reviewer start?
`Select/doc.js`

#### What testing has been done on this PR?
- manual testing in local project
- added tests for:
  - apply new documentation definition: select.options.container
  - confirm backwards compatibility support for legacy select.options.box
  - apply both select.options.container and select.options.box styling when both are present; select.options.container is prioritized and applied if conflict between the two.

#### Any background context you want to provide?
https://github.com/grommet/grommet/issues/3183

#### What are the relevant issues?
- https://github.com/grommet/grommet/issues/3183
- https://github.com/grommet/grommet/pull/3184#issuecomment-508264393

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Comment by ShimiSun** [comment]
@ericsoderberghp I'm inclining towards `container` over `box`, thoughts?

**Comment by ShimiSun** [comment]
Thanks for catching this issue and making a contribution!
We are inclining towards `container` rather than `box` since it is more semantically correct.
Changing the code to support it may cause a breaking change so we will need to support both `container` and `box` when applying the theme to avoid it.
So base.js will be something like `select: { options: { container: { ... } } }`, but the code that applies the theme will support `box || container`, let me know what you think and if that is something you'll be able to work on. thanks again.

**Comment by halocline** [comment]
Not a problem. Can and will do. Originally opted for this approach because of the breaking change, but can implement your suggestion.

**Comment by halocline** [comment]
@ShimiSun --- Modified the documentation to reference preferred terminology of `container` instead of `box`. Added backwards compatibility supporting both `select.options.container` and `select.options.box`. Added tests to confirm new definition and backwards compatibility support.

File: src/js/components/Select/SelectContainer.js [hunk 1]
```diff
@@ -359,6 +359,10 @@ class SelectContainer extends Component {
 
     const customSearchInput = theme.select.searchInput;
     const SelectTextInput = customSearchInput || TextInput;
+    const selectOptionsStyle = {
+      ...theme.select.options.box,
+      ...theme.select.options.container,
```
Comment (ShimiSun) [inline src/js/components/Select/SelectContainer.js]: what will happen if we'll have same props defined on both `theme.select.options.box` & `theme.select.options.container`?

File: src/js/components/Select/__tests__/Select-test.js [hunk 2]
```diff
@@ -416,4 +416,112 @@ describe('Select', () => {
     fireEvent.click(selectButton);
     expect(selectButton).toHaveStyleRule('background', 'purple');
   });
+
+  test(`renders styled select options backwards compatible with legacy
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/Select-test.js]: Nice work on the tests!

File: src/js/components/Select/SelectContainer.js (hunk 1 above)
Comment (halocline) [reply src/js/components/Select/SelectContainer.js]: `select.options.container` will "win". I have included a test for this scenario.

**Comment by ShimiSun** [comment]
Nice work!! thanks for following up with the feedback and getting to this end result!
Thanks for the contribution 🎉

**Pull request merged**